### PR TITLE
consider repokey w/o passphrase == unencrypted

### LINKED ISSUE #2169: Interactive repository swap
From #2160 

Note that when using BORG_PASSPHRASE the attacker cannot swap the *entire*
repository against a new repository with e.g. repokey mode and no passphrase,
because Borg will abort access when BORG_PASSPRHASE is incorrect.

However, interactively a user might not notice this kind of attack
immediately, if she assumes that the reason for the absent passphrase
prompt is a set BORG_PASSPHRASE.

---

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/42112412-interactive-repository-swap)

note: bounty is for checking /fixing all mentioned "repo swap" issues.

### PR DESCRIPTION
**Pull request by enkore**
Fixes #2169

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -128,6 +128,15 @@ The best check that everything is ok is to run a dry-run extraction::
...
 
+Version 1.1.0b6 (unreleased)
+----------------------------
+
+Compatibility notes:
+
+- Repositories in a repokey mode with a blank passphrase are now treated
+  as unencrypted repositories for security checks
+  (e.g. BORG_UNKNOWN_UNENCRYPTED_REPO_ACCESS_IS_OK).
+
```
Comment (ThomasWaldmann) [inline docs/changes.rst]: Maybe also list it below "Fixes:" as a security-related fix?

While it is not strictly a borg bug, a user could be tricked easily enough with it.

Comment (enkore) [inline docs/changes.rst]: _Compatibility Notes_ is usually at the top, even before _Fixes_. Perhaps something to consider for condensed 1.1 released notes?

Comment (ThomasWaldmann) [inline docs/changes.rst]: My comment was not just about "having it at the top", but about explicitly saying it is security related fix.

Comment (enkore) [inline docs/changes.rst]: Ah. Perhaps it'd still be better in that case to amend a succinct description of the issue/scenario to the changelog entry, not necessarily duplicating the text.

Comment (enkore) [inline docs/changes.rst]: What do you think about this?

> - Repository in a  ...
>
> Previously it was possible to be tricked into using a repository (repokey,
> blank passphrase) swapped for an encrypted repository, if one assumed
> that BORG_PASSPHRASE was set when it was not. This does not affect scripts
> etc. using BORG_PASSPHRASE.
>

Comment (ThomasWaldmann) [inline docs/changes.rst]: "in a ..." - what do you mean?

Hmm, is the order of arguments of "swapped for" correct here?
Maybe use an easier to understand form.

Comment (enkore) [inline docs/changes.rst]: "Repository in a ..." = Existing changelog entry (see above)

**Pull request merged**